### PR TITLE
[FIX] l10n_it_edi: import company to company invoice

### PR DESCRIPTION
**Pull request by lordkrandel**
As we imported invoices from the IAP proxy we refused files when the same filename was already present in the database.

Import invoices sent by Company A to Company B on the same Odoo database was therefore impeded. The CRON would try to import the Company B's vendor bill with the same attachment's filename as the Company A's invoice and reject the file.

Now we fill in the `company_id` field on the attachment and we search for attachments which belong to the company we're importing for, so the case is covered.

An import test has been added.

Ticket [link](https://www.odoo.com/web#model=project.task&id=3673508)
opw-3673508

### CODE REVIEW

**Comment by h4818** [comment]
could the same proxy_user not be used in multiple companies?

**Comment by lordkrandel** [comment]
There's one (or more) proxy_user(s) per company.

**Comment by lordkrandel** [comment]
ping @h4818 

Here is further explanation:

- Proxy Users for Italian Localization are "registrations" to the IAP server (or the Test IAP server). They are created automatically when the user allows Odoo to handle their invoices through the EDI, ticking the corresponding box in the Settings.

  image

  The company field is also hidden so the user doesn't get to ever modify it, or reassign it.
  image

- When downloading files, we loop over Proxy Users registrations on our Odoo instance to download from the IAP proxy server the files for each company (one per proxy_user).
  Before we save files, we check the file is already there. Problem is, if Company A invoices Company B (2 companies same Odoo), the Tax Agency takes the file from Company A and re-routes it back to the IAP Proxy Server with the same name for Company B. It won't be imported, since it is already present with the same name for the other company. So when we check for the filename, we should make sure that we filter by company.

File: addons/l10n_it_edi/tests/test_edi_import.py [hunk 1]
```diff
@@ -96,6 +99,51 @@ def mock_commit(self):
...
+        attachments = self.env['ir.attachment'].search([
+            ('name', '=', 'IT01234567890_FPR02.xml'),
+            ('res_model', '=', 'account.move'),
+            ('res_field', '=', 'l10n_it_edi_attachment_file'),
+        ])
+        self.assertEqual(2, len(attachments))
+
+        found_moves = self.env['account.move'].browse(attachments.mapped("res_id"))
+        self.assertEqual(2, len(set(found_moves.ids)))
+        self.assertEqual(set(found_moves.mapped("company_id.name")), {"company_1_data", "company_2_data"})
```
Comment (h4818) [inline addons/l10n_it_edi/tests/test_edi_import.py:145]: Are we not testing something that was set explicitly?
I suggest checking only the attachment & move we didn't create

**Review by h4818 — COMMENTED** [review]
gg @lordkrandel 
The fix LGTM, just a few comments on the test.

File: addons/l10n_it_edi/tests/common.py [hunk 2]
```diff
@@ -100,6 +102,9 @@ def setUpClass(cls, chart_template_ref='it'):
 
         cls.module = 'l10n_it_edi'
 
+    def _get_uuid(self):
+        return str(uuid.uuid4())
```
[the remark this answers is not in the document]
Comment (lordkrandel) [reply addons/l10n_it_edi/tests/common.py:106]: I'll leave it all like it is now, we'll refactor those when we'll need it.

File: addons/l10n_it_edi/tests/test_edi_import.py [hunk 3]
```diff
@@ -96,6 +99,51 @@ def mock_commit(self):
...
+        filename = 'IT01234567890_FPR02.xml'
+
+        invoice = self.env['account.move'].with_company(other_company).create({
+            'move_type': 'out_invoice',
+            'invoice_line_ids': [
+                Command.create({
+                    'name': "something not price included",
+                    'price_unit': 800.40,
+                    'tax_ids': [Command.set(self.company_data['default_tax_sale'].ids)]
+                })]})
```
[the remark this answers is not in the document]
Comment (lordkrandel) [reply addons/l10n_it_edi/tests/test_edi_import.py:116]: We tried many times, but everytime in the localization tests we need something a little different than the time before, and the helper function ends up being a mess. 
I'm in favour of creating test-data for each localization just like we have demo-data, but that's another matter.

**Comment by h4818** [comment]
@robodoo r+

**Pull request closed without merging**